src/client/client.socket.ts

```diff
@@ -54,6 +55,11 @@ export class CommandResponseStream extends Duplex {
     this.isAuthenticated = false;
   };
 
+  // Probably triggered by Duplex class
+  _destroy() {
```

Comment:
TcpClient is a RawClient which is a `CommandResponseStream` which wrap the socket. Destroy function is called on `CommandResponseStream` why it should be called on the socket. This function allows to forward the signal to the socket.
`destroy` is not allowed because it does not respect the interface of `Duplex` and `_destroy` works. I suppose `destroy` triggers `_destroy` in `Duplex`.